### ISSUE TITLE
property name not checked in validity

### ISSUE THREAD

**Issue by tong67** [pimatic/pimatic#696]
I just started exploring pimatic. After following the HOWTO intro I run into problems with the homeduino. After many experiments I finally was able to find my mistake. I used switch6 with properties "systemcode " and "programcode ". Notice the space added to the names. No error were reported on any level. When I found my mistake I tried with "foo" instead of "systemcode". Also that was accepted without any error but of course no working system.

Is it possible to add checking on correctness/support of used property names?
I have no idea whether this is a plugin specific item or a framework item.

**Comment by mwittig** [comment]
Agree. Validity checking is still something which needs to be implemented in many places throughout the pimatic codebase. Regarding homeduino is also confusing as the various switching protocols have varying properties.

Regarding your question:
> Is it possible to add checking on correctness/support of used property names?

Yes, certainly. To start with it should be implemented as part of the rfcontroljs package so that checking is available for the homeduino protocols. However, it is also required in the pimatic core.

**Comment by Icesory** [comment]
done for homeduino
https://github.com/pimatic/pimatic-homeduino/commit/01efdcec9aca0894dc31b251dc1bcb65000e1cb6
